tests/jballerina-unit-test/src/test/resources/test-src/statements/ifelse/type-guard.bal

```diff
@@ -760,6 +760,19 @@ function testTypeNarrowingForIntersectingDirectUnion_1() returns boolean {
     return false;
 }
 
+type CyclicComplexUnion int|CyclicComplexUnion[]|object {};
```

Comment:
Can we add negative tests also? Where there's no intersection.

Reply:
done

Reply:
Can we include a negative compilation test too?

Reply:
added to `type-guard-negative.bal`